src/rpc/server.rs

```diff
@@ -289,18 +292,46 @@ async fn execute_tx_impl(
     }
 
     match gas_station
-        .execute_transaction(reservation_id, tx_data, user_sig)
+        .execute_transaction(reservation_id, tx_data, user_sig, options.clone())
         .await
     {
-        Ok(effects) => {
+        Ok(mut tx_block_response) => {
```

Comment:
I think you might be able to use a more sophisticated pattern, like below to make sure that in this codepath, you always have effects, and avoid the unwrap.

```suggestion
        Ok(mut tx_block_response @ SuiTransactionBlockResponse { effects: Some(ref effects), .. }) => {
```

Then you need to add another match arm at the top level to catch the case where `effects: None`.

Failing that, use a `let Some(effects) = &tx_block_response.effects else { ... }` block at the top here, to avoid the unwrap below.